### PR TITLE
[TTS] Fix TTS audio preprocessing bugs

### PR DESCRIPTION
**Pull request by rlangman**
# What does this PR do ?

Fix a few bugs in TTS preprocess_audio.py and MelSpectrogramFeaturizer.

**Collection**: [TTS]

# Changelog 
- Add additional config to mel featurizer to make it more consistent with past TTS configs
- Add check and unit test for normalize_volume when audio input is empty array.
- Modify preprocess script to avoid error when resample_rate=0.
- Modify preprocess script to round duration to 2 decimals
- Move normalize_volume unit tests to correct test class, following an earlier refactoring.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

File: scripts/dataset_processing/tts/audio_processing/preprocess_audio.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def _process_entry(
     original_duration = librosa.get_duration(filename=audio_path)
     output_duration = librosa.get_duration(filename=output_path)
 
-    entry["duration"] = output_duration
+    entry["duration"] = round(output_duration, 2)
```
[the remark this answers is not in the document]
Comment (rlangman) [reply scripts/dataset_processing/tts/audio_processing/preprocess_audio.py:143]: Without rounding you end up with a large ~10 decimal number in the manifest, which is unpleasant to look at. The "duration" field in the public HiFi-TTS dataset is rounded to 2 decimal places (we could do 3 if we wanted more precision).

**Review by redoctopus — APPROVED** [review]
LGTM!
